### ISSUE TITLE
Shared vehicle orders

### ISSUE THREAD

**Issue by LeftofZen** [OpenLoco/OpenLoco#1440]
**Describe the solution you'd like**
If you have a large vehicle network and all your vehicles have the same order (via cloning the vehicle, for example), then if you need to update that route, you need to manually update it for every single vehicle. Having a shared order would solve this issue as all vehicles would reference the single shared order and updating that shared order would 'apply' the changes to all vehicles linked to that shared order.

Care must be taken in implementation - each vehicle still needs to store their own 'local' set of orders to maintain where in the order list they are up to and what the next waypoint/station is. Additionally when a shared order is modified, how should the linked vehicles respond? Maintaining where they are in the order list is probably the safest bet (as opposed to restarting from the top of the list).

A new window may need to be added to facilitate creating, editing and deleting these shared orders. However this isn't necessary and you could get away with not having the shared orders window by simply 'linking' trains and automatically sharing the orders. For example you set up orders on train 1, and on train 2 you say "follow train 1's orders". In code, a shared order as described above is created, and both train 1 and train 2 reference it. Editing either the orders on train 1 or train 2 are actually edits to the shared order, and so the other trains linked to this order are updated automatically. Vehicles would then need a 'detach from shared order' button to remove the link.

Multiple vehicle types should be able to share the same shared order. Eg trains and buses and trucks could all use the same single shared order if all 3 modes of transport have stations in the shared station id.

**Describe alternatives you've considered**
None that I can think of.

**Additional context**
OpenTTD has shared orders, so this implementation would be similar to theirs.
This is basically adding pointer-semantics to vehicle orders.

**Comment by duncanspumpkin** [comment]
At present all that is required to make this work is that the vehicle sell function is reworked to check to make sure the order table is not deleted on sell. Other than that everything is already setup to share orders.

**Comment by LeftofZen** [comment]
Interesting, I might give it a go then

**Comment by duncanspumpkin** [comment]
Yeah sure thats a given i think when we did the clone function originally it accidentally shared orders.
